File: 06_pattern_matching/python/fizzbuzz.py
if __name__ == "__main__":
    # not possible as lambda
    def if_buzz(x):
        if x % 15 == 0:
            print("FizzBuzz")
        elif x % 3 == 0:
            print("Fizz")
        elif x % 5 == 0:
            print("Buzz")
        else:
            print(x)

    def fizz_buzz(x):
        match x % 3, x % 5:
            case 0, 0: print("FizzBuzz"),
            case 0, _: print("Fizz"),
            case _, 0: print("Buzz"),
            case _, _: print("{}", x)

    list(map(fizz_buzz, range(1, 24)))

File: 06_pattern_matching/rust/fizzbuzz.rs
fn main() {
    fn if_buzz(x: i32) {
        if x % 15 == 0 {
            println!("FizzBuzz")
        } else if x % 3 == 0 {
            println!("Fizz")
        } else if x % 5 == 0 {
            println!("Buzz")
        } else {
            println!("{}", x)
        }
    }

    let fizz_buzz = |x| match (x % 3, x % 5) {
        (0, 0) => println!("FizzBuzz"),
        (0, _) => println!("Fizz"),
        (_, 0) => println!("Buzz"),
        (_, _) => println!("{}", x),
    };

    (1..24).into_iter().for_each(fizz_buzz);
}
